tests/ctst/common/common.ts

```diff
@@ -85,3 +90,91 @@ Given('an object {string} that {string}', async function (this: Zenko, objectNam
         await putObject(this, objectName);
     }
 });
+
+When('the user tries to perform the current S3 action on the bucket {int} times with a {int} ms delay',
+    async function (this: Zenko, numberOfRuns: number, delay: number) {
+        this.setAuthMode('test_identity');
+        const action = {
+            ...this.getSaved<ActionPermissionsType>('currentAction'),
+        };
+        if (action.action.includes('Version') && !action.action.includes('Versioning')) {
+            action.action = action.action.replace('Version', '');
+            this.addToSaved('currentAction', action);
+        }
+        for (let i = 0; i < numberOfRuns; i++) {
+            // For repeated WRITE actions, we want to change the object name
+            if (action.action === 'PutObject') {
+                this.addToSaved('objectName', `objectrepeat-${Utils.randomString()}`);
+            } else if (action.action === 'CopyObject') {
+                this.addToSaved('copyObject', `objectrepeatcopy-${Utils.randomString()}`);
+            }
+            await runActionAgainstBucket(this, this.getSaved<ActionPermissionsType>('currentAction').action);
+            if (this.getResult().err) {
+                // stop at any error, the error will be evaluated in a separated step
+                return;
+            }
+            await Utils.sleep(delay);
+        }
+    });
+
+Then('the API should {string} with {string}', function (this: Zenko, result: string, expected: string) {
+    this.cleanupEntity();
+    const action = this.getSaved<ActionPermissionsType>('currentAction');
+    switch (result) {
+    case 'success':
```

Comment:
Yes good catch, tests were not properly checking results in the happy scenarios, now it passes (got some issues for the deletes, because of bad values in the feature file, let's see the results now)